### DISCUSSION TITLE
Images from timeline variables in survey-text

### DISCUSSION THREAD

**Discussion by aaron-cousins** [jspsych/jsPsych#2933] [Q&A] [answered]
Hi, 
I'm trying to build an artificial language learning experiment where participants see a novel object, give it a name, then get feedback and eventually pick up the artificial language through exposure. But I'm having trouble getting the images to show up from timeline variables using the survey-text plugin. I get a file not found error. The feedback trials which use the image-button-response plugin work fine, so it's not an issue with pre-loading or declaring the timeline variables. 

Here's my current code: 

```js
  var response_trial = {
    type: jsPsychSurveyText,
    preamble: '<img src="' +jsPsych.timelineVariable('stimulus')+ '">',
    questions: [{prompt: 'Enter the name of the object:'}],
    data: {
        stimulus: jsPsych.timelineVariable('stimulus'),
        correct_response: jsPsych.timelineVariable('name'),
        trial_type: 'response'
    }
  };
```
I think it's because I'm putting an array in the image tags, but I'm not sure how to fix it without creating a different response trial for each stimulus. 

Thanks.

**Comment by jodeleeuw** [comment, marked as answer]
Hi @aaron-cousins,

Because you are using a function (`jsPsych.timelineVariable('stimulus')`) in combination with the string to make the `img` tag, you'll need to use a [dynamic parameter](https://www.jspsych.org/7.3/overview/dynamic-parameters/). 

```js
 preamble: function() { return '<img src="' +jsPsych.timelineVariable('stimulus')+ '">' },
```

**Reply by aaron-cousins** [reply]
Thank you!
